mini-projects/00-jq-rs/src/processor.rs

```diff
@@ -0,0 +1,253 @@
+            "." => output.push((Commands::Filter(FilterType::Identity), String::new())),
+            ".[]" => output.push((Commands::ArrayIterator, String::new())),
+            //now we want to search for commands like delete[], delete<>,array index,."key" and array slice
+            //most convient way to do this I found was to use REGEX
+            _ => {
+                //no need for errorhandling as the regex is not userfacing
+                let delete_key_regex = Regex::new(r"del\(\.(.*?)\)").unwrap();
+                let delete_index_regex = Regex::new(r"del\[\.(.*?)\]").unwrap();
+                let array_slice_regex = Regex::new(r"\.\[([^\]]*:[^\]]*)\]").unwrap();
+                let array_index_regex = Regex::new(r"\.\[(.*?)\]").unwrap();
+                let object_regex = Regex::new(r"\.(\w+)").unwrap();
+                match delete_key_regex.captures(string) {
+                    //this means we found a delete regex with a key needle
+                    Some(needle) => output.push((Commands::Delete, String::from(&needle[1]))),
+                    None => {
+                        //check if we have a delete with an index
+                        match delete_index_regex.captures(string) {
+                            Some(needle) => {
+                                output.push((Commands::Delete, String::from(&needle[1])))
+                            }
+                            None => {
+                                //Check for an array slice
+                                match array_slice_regex.captures(string) {
+                                    Some(needle) => output.push((
+                                        Commands::Filter(FilterType::ArraySlice),
+                                        String::from(&needle[1]),
+                                    )),
+                                    None => {
+                                        //check if it is an array index
+                                        match array_index_regex.captures(string) {
+                                            Some(needle) => output.push((
+                                                Commands::Filter(FilterType::ArrayIndex),
+                                                String::from(&needle[1]),
+                                            )),
+                                            None => {
+                                                //finally we can check if it is an object index, if it isn't this than the input was invalid so the code will panic
+                                                let needle = object_regex.captures(string).unwrap();
+                                                output.push((
+                                                    Commands::Filter(FilterType::ObjectIdentity),
+                                                    String::from(&needle[1]),
+                                                ))
```

Comment:
lol this verbosely nesting is kinda gross, in this case, I probably would have opted for if and elifs